packages/manager/src/features/Account/Quotas/Quotas.test.tsx

```diff
@@ -0,0 +1,171 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { fireEvent, screen, waitFor } from '@testing-library/react';
+import * as React from 'react';
+
+import { regionFactory } from 'src/factories';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+
+import { Quotas } from './Quotas';
+
+const queryMocks = vi.hoisted(() => ({
+  useFlags: vi.fn().mockReturnValue({}),
+  useGetLocationsForQuotaService: vi.fn().mockReturnValue({}),
+  useGetRegionsQuery: vi.fn().mockReturnValue({}),
+  useQuotasQuery: vi.fn().mockReturnValue({}),
+}));
+
+vi.mock('src/queries/regions/regions', () => ({
+  useRegionsQuery: queryMocks.useGetRegionsQuery,
+}));
+
+vi.mock('src/hooks/useFlags', () => {
+  const actual = vi.importActual('src/hooks/useFlags');
+  return {
+    ...actual,
+    useFlags: queryMocks.useFlags,
+  };
+});
+
+vi.mock('src/queries/quotas/quotas', () => ({
+  useQuotasQuery: queryMocks.useQuotasQuery,
+}));
+
+vi.mock('./utils', () => ({
+  useGetLocationsForQuotaService: queryMocks.useGetLocationsForQuotaService,
+}));
+
+describe('Quotas', () => {
+  beforeEach(() => {
+    queryMocks.useGetLocationsForQuotaService.mockReturnValue({
+      isFetchingRegions: false,
+      regions: [
+        regionFactory.build({ id: 'global', label: 'Global (Account level)' }),
+      ],
+    });
+    queryMocks.useGetRegionsQuery.mockReturnValue({
+      data: [
+        regionFactory.build({ id: 'global', label: 'Global (Account level)' }),
+        regionFactory.build({ id: 'us-east', label: 'Newark, NJ' }),
+      ],
+      isFetching: false,
+    });
+    queryMocks.useQuotasQuery.mockReturnValue({
+      data: { data: [] },
+      isFetching: false,
+      refetch: vi.fn(),
+    });
+  });
+
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        retry: false,
+      },
+    },
+  });
+
+  it('renders the component with initial state', () => {
+    const { getByText } = renderWithTheme(
+      <QueryClientProvider client={queryClient}>
+        <Quotas />
+      </QueryClientProvider>
+    );
```

Comment:
Our `renderWithTheme` function allows you to pass a custom `queryClient` in the options if needed. We may want to do that here

```suggestion
    const { getByText } = renderWithTheme(<Quotas />, { queryClient });
```

Reply:
always forget! done